core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeKinesisStreamSink.java

```diff
@@ -0,0 +1,210 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.realtime.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.realtime.DataWriter;
+import co.cask.cdap.etl.api.realtime.RealtimeContext;
+import co.cask.cdap.etl.api.realtime.RealtimeSink;
+import co.cask.hydrator.plugin.common.Properties;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.PutRecordRequest;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+
+import static co.cask.hydrator.plugin.common.KinesisUtil.createAndWaitForStream;
+
+/**
+ * A {@link RealtimeSink} that writes data to a Amazon Kinesis stream.
+ * If Kinesis Stream does not exist, it will be created using properties provided with this sink.
+ */
+@Plugin(type = RealtimeSink.PLUGIN_TYPE)
+@Name("KinesisSink")
+@Description("Real-time sink that outputs to a specified AWS Kinesis stream.")
+public class RealtimeKinesisStreamSink extends RealtimeSink<StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RealtimeKinesisStreamSink.class);
+  private final KinesisConfig config;
+  private AmazonKinesisClient kinesisClient;
+
+  public RealtimeKinesisStreamSink(KinesisConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.name),
+                                "Stream name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessKey),
+                                "Access Key should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessSecret),
+                                "Access Key secret should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.partition),
+                                "Partition name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.bodyField),
+                                "Input field should be non-null, non-empty.");
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    validateInputSchema(inputSchema);
+  }
+
+  private void validateInputSchema(@Nullable Schema inputSchema) {
+    if (inputSchema != null) {
```

Comment:
what happens when ``inputSchema`` is null?

Also, to reduce indentation, we prefer:

```
if (inputSchema == null) {
  return;
}
// Rest of the logic ...
```